src/status_im/contexts/centralized_metrics/core.cljs

```diff
@@ -0,0 +1,88 @@
+(ns status-im.contexts.centralized-metrics.core
+  (:require
+    [legacy.status-im.utils.build :as build]
+    [native-module.core :as native-module]
+    [re-frame.interceptor :as interceptor]
+    [react-native.platform :as platform]
+    [taoensso.timbre :as log]
+    [utils.re-frame :as rf]))
+
+(defn user-journey-event
+  [action]
+  {:metric
+   {:eventName  "user-journey"
+    :platform   platform/os
+    :appVersion build/app-short-version
+    :eventValue {:action action}}})
+
+(def app-started-event "app-started")
+(def new-to-status-event "clicked-on-new-to-status")
+(def navigate-to-create-profile-event "navigate-to-create-profile")
+(def communities-tab-clicked "communities-tab-clicked")
+(def wallet-tab-clicked "wallet-tab-clicked")
+(def chats-tab-clicked "chats-tab-clicked")
+
+(defn track-view-id-event
+  [view-id]
+  (case view-id
+    :communities-stack (user-journey-event communities-tab-clicked)
+    :chats-stack       (user-journey-event chats-tab-clicked)
+    :wallet-stack      (user-journey-event wallet-tab-clicked)
+    nil))
+
+(defn tracked-events
+  [[event-name second-parameter]]
+  (case event-name
+    :onboarding/navigate-to-create-profile
+    (user-journey-event navigate-to-create-profile-event)
+
+    :profile/get-profiles-overview-success
+    (user-journey-event app-started-event)
+
+    :set-view-id
+    (track-view-id-event second-parameter)
+
+    nil))
+
+(defn should-push-event?
+  [db]
+  (or (not (:centralized-metrics/user-confirmed? db))
+      (:centralized-metrics/enabled? db)))
+
+(defn push-event
+  [e]
+  (native-module/add-centralized-metric
+   e
+   #(log/info "pushed metric" % e)))
+
+(defn toggle
```

Comment:
We could directly call `toggle-centralized-metrics` because this `toggle` function is only wrapping `enabled` in a map. Or we can also change `native-module/toggle-centralized-metrics` signature to be more useful. `toggle-centralized-metrics` is still a Clojure layer, so the API we expose can be more useful instead of having one layer exposing a function that needs another wrapper. I hope that makes sense 🤷🏼